### ISSUE TITLE
missing param in ServiceBusException.Message ?

### ISSUE THREAD

**Issue by SimonCropp** [Azure/azure-service-bus-dotnet#329]
`StringUtility.FormatInvariant` is

```
        public static string FormatInvariant(this string format, object arg0)
        {
            return string.Format(CultureInfo.InvariantCulture, format, arg0);
        }
```
and is used in `ServiceBusException.Message` with 

```
return StringUtility.FormatInvariant("{0}, ({1})", this.ServiceBusNamespace);
```

but `{0}, ({1})` has two params but is only passed one value.

Am i missing something?

**Comment by SeanFeldman** [comment]
https://github.com/Azure/azure-service-bus-dotnet/blob/10fb579c967200563c108b4e4282de5ead79730e/src/Microsoft.Azure.ServiceBus/ServiceBusException.cs#L58-L70

I suspect the idea was to call with with `baseMessage`.

```c#
return StringUtility.FormatInvariant("{0}, ({1})", this.ServiceBusNamespace, baseMessage);
```

**Comment by SimonCropp** [comment]
also since `string.Format` already has non array based overload for perf, `StringUtility.FormatInvariant` seems redundant. i also suspect the added layer of indirection contributed to this bug.

**Comment by nemakam** [comment]
Lets make it the other way
```c#
return StringUtility.FormatInvariant("{0}, ({1})", baseMessage, this.ServiceBusNamespace);
```

**Comment by SimonCropp** [comment]
@nemakam can u articulate the value of the single use helper method that just calls one other method?

**Comment by nemakam** [comment]
> @nemakam can u articulate the value of the single use helper method that just calls one other method?

Which piece of code are you referring to exactly here?

**Comment by nemakam** [comment]
Without knowing the context, the main thing I can think of is readability. Who wants to read a monolithic 100 lines of code if that can be split into 7 different methods? Going through the code and understanding it becomes quite easier.

**Comment by nemakam** [comment]
My bad.. I assumed we are on the other PR thread.
If you are talking about `StringUtility.FormatInvariant`, then I am not inclined towards any particular approach. I'm okay with either having this method or taking it off. Maybe reduces the code by 20 characters overall?

**Comment by clemensv** [comment]
@nemakam Is there any more status on this issue?

**Comment by nemakam** [comment]
Fixed in #347
